### PR TITLE
refactor(nocodb): backend

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

- `created_at` and `updated_at` are auto-populated - we don't need to specify them in models.
- add `PagedResponseImpl`
- map service functions
- remove outdated todo tags / empty import lines
- load lodash packages individually 
- revise response messages for which don't return data 
- revise api docs
- unifying listing types. The following APIs would have the following response format
```
{
  "list": [{...}, {...}],
  "pageInfo": {...}
}
```

Full List:
  - /api/v1/db/meta/projects/:projectId/api-tokens
  - /api/v1/db/meta/audits/comments
  - /api/v1/db/meta/views/:viewId/filters
  - /api/v1/db/meta/filters/:filterParentId/children
  - /api/v1/db/meta/hooks/:hookId/filters
  - /api/v1/users
  - /api/v1/db/meta/views/:viewId/sorts/
  - /api/v1/db/meta/tables/:tableId/views
  - /api/v1/db/meta/tables/:tableId/share
  - /api/v1/db/meta/views/:viewId/columns/
  - /hooks/:hookId/filters/
  - /hooks/:hookId/filters/:filterParentId/children

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/account/UserList.vue [hunk 1]
```diff
@@ -59,7 +61,7 @@ const updateRole = async (userId: string, roles: Role) => {
   try {
     await api.orgUsers.update(userId, {
       roles,
-    } as unknown as UserType)
+    } as OrgUserReqType)
```
Comment (pranavxc) [inline packages/nc-gui/components/account/UserList.vue:64]: We can even remove the `as OrgUserReqType`

File: packages/nc-gui/composables/useKanbanViewStore.ts [hunk 2]
```diff
@@ -176,9 +176,16 @@ const [useProvideKanbanViewStore, useKanbanViewStore] = useInjectionState(
...
             where,
           })
-        : await fetchSharedViewData({ sortsArr: sorts.value, filtersArr: nestedFilters.value, offset: params.offset })
+        : await fetchSharedViewData({
+            ...{ where: xWhere.value },
+            ...params,
+            sortsArr: sorts.value,
+            filtersArr: nestedFilters.value,
+            offset: params.offset,
+            where,
```
Comment (pranavxc) [inline packages/nc-gui/composables/useKanbanViewStore.ts:185]: `...{ where: xWhere.value }` (L180) will get overwritten by `where,` (L185)

File: packages/nc-gui/components/account/UserList.vue (hunk 1 above)
Comment (wingkwong) [reply packages/nc-gui/components/account/UserList.vue:64]: Type 'Role' is not assignable to type '"org-level-creator" | "org-level-viewer" | undefined'.

**Review by pranavxc — APPROVED** [review]
List down the APIs whose response had been modified for the release note.

**Pull request merged**